main.go

```diff
@@ -30,7 +30,10 @@ import (
 	"os"
 )
 
-func oidcAuthBuilder(options *auth.Options) (auth.AuthenticationStrategy, *auth.Options, error) {
+func oidcAuthBuilder(options *auth.Options, clientCredentialsFlow bool) (auth.AuthenticationStrategy, *auth.Options, error) {
+	if clientCredentialsFlow {
+		return oidc.NewClientCredentialsStrategy(options)
```

Comment:
both strategies have the same code base with 1 line diference - e should try to refactor to avoid the duplication